src/components/application_manager/src/message_helper/message_helper.cc

```diff
@@ -738,10 +738,13 @@ void MessageHelper::SendDeleteChoiceSetRequest(smart_objects::SmartObject* cmd,
   using namespace smart_objects;
 
   SmartObject msg_params = SmartObject(smart_objects::SmartType_Map);
-
   msg_params[strings::app_id] = application->app_id();
   msg_params[strings::type] = hmi_apis::Common_VRCommandType::Choice;
-  msg_params[strings::grammar_id] = (*cmd)[strings::grammar_id];
+
+  if (cmd->keyExists(strings::grammar_id) &&
+      (*cmd)[strings::grammar_id].asInt() != -1) {
+    msg_params[strings::grammar_id] = (*cmd)[strings::grammar_id];
```

Comment:
```suggestion
  if (!cmd->keyExists(strings::grammar_id) ||
      (*cmd)[strings::grammar_id].asInt() == -1) {
    return;
```
Same here, should just skip over it